hedera-node/src/main/java/com/hedera/services/fees/calculation/AutoRenewCalcs.java

```diff
@@ -137,7 +146,8 @@ long rbUsedBy(MerkleAccount account) {
 				.setCurrentKey(asKeyUnchecked(account.getAccountKey()))
 				.setCurrentlyHasProxy(account.getProxy() != null)
 				.setCurrentMemo(account.getMemo())
-				.setCurrentNumTokenRels(account.tokens().numAssociations())
+				.setCurrentNumTokenRels(
```

Comment:
Two things here:
1. We likely want to instead maintain a count of associated tokens in the `MerkleAccountState` leaf---`MerkleMap.get()` calls are quite expensive, so doing 100's or 1000's of them here will really add up.
2. ⚠️ Even if we are 99.999% confident there is no way for a linked list of associations to have a loop, imo we should not make an essentially unbounded call to a method like `getAssociatedTokens()`...making 2B consecutive calls to `MerkleMap.get()` would probably bring down the network!

Reply:
I did think about saving the count on MerkleAccountState rather than traversing the list just for the size but held it back. Will go down that road now.